src/Language/PureScript/CoreFn/ToJSON.hs

```diff
@@ -94,10 +94,18 @@ properNameToJSON :: ProperName a -> Value
 properNameToJSON = toJSON . runProperName
 
 qualifiedToJSON :: (a -> Text) -> Qualified a -> Value
-qualifiedToJSON f (Qualified mn a) = object
-  [ T.pack "moduleName"   .= maybe Null moduleNameToJSON mn
-  , T.pack "identifier"   .= toJSON (f a)
-  ]
+qualifiedToJSON f (Qualified qb a) =
+  case qb of
+    ByModuleName mn -> object
+      [ T.pack "moduleName" .= moduleNameToJSON mn
+      , T.pack "identifier" .= toJSON (f a)
+      ]
+    BySourceSpan ss -> object
+      -- TODO: Should `sourceSpanXJSON` encode this information as well?
+      [ T.pack "spanName"   .= toJSON (spanName ss)
```

Comment:
Is exposing the `spanName` here potentially a bad idea? Should we aim to make a default for this like `"<local>"`?

Reply:
I'm not sure.

Reply:
@nwolverson would this be something that affects the purerl backend?

Reply:
Coming back to this question, what does `spanName` store? The only reference I found using a cursory glance was https://github.com/purescript/purescript/blob/f9f75f43bd1bdbc1e866fdba876582385e3f4c65/tests/Language/PureScript/Ide/RebuildSpec.hs#L75

Or another way to think about this question. If this was not added, could it be added in the future without incurring a breaking change?

I don't know enough about `spanName` to make a judgement here.

Reply:
`spanName` usually stores the absolute(?) file path to the source file alongside the source spans, and in certain cases may not contain information at all (i.e. null source spans, which have now been repurposed to mean "generated by the compiler"). Being an absolute file path may or may not be a good thing if there's some sensitive information located in the path name, although I can't say for certain since I don't have that constraint.

Since `Qualified` names now rely on source spans to be disambiguated through `BySourceSpan`, we need to stabilize `spanName` somehow.

Reply:
> Being an absolute file path may or may not be a good thing if there's some sensitive information located in the path name, although I can't say for certain since I don't have that constraint.

Ah... I see. I don't know how often this occurs, but I'd leaned towards not worrying about this. Or to state it differently, I don't have this constraint either.

> Since Qualified names now rely on source spans to be disambiguated through BySourceSpan, we need to stabilize spanName somehow.

Sounds like this is the next course of action then?

Reply:
If privacy is a concern, could `spanName` be changed to store only the relative reference for `BySourceSpan` values?

Reply:
Ok, I think the absolute paths come from how the `purs` binary is invoked and given the source files. When I do something like `purs compile Main.purs`, it sets the `modulePath` for a module to an absolute path, although when I do `purs compile src/Main.purs`, the `modulePath` just shows as relative.

I've only inspected the output of this change using the former command on a temporary folder, which explains the absolute paths. Would it be better to extract this to a separate PR?

Reply:
> Would it be better to extract this to a separate PR?

Sure!